packages/next/shared/lib/router/router.ts

```diff
@@ -936,7 +937,7 @@ export default class Router implements BaseRouter {
     if (
       !(options as any)._h &&
       this.onlyAHashChange(cleanedAs) &&
-      !localeChange
+      prevLocale === this.locale
```

Comment:
It looks like `localeChange` isn't being updated correctly, can we keep this as `!localeChange` and then move `localeChange` to be set on line 930 instead of line 820 and to be set to `prevLocale !== this.locale`?

Reply:
Updated